### PR TITLE
New parameters for the checkout functions

### LINKED ISSUE #26: Additional checkout parameters
The JavaScript SDK now supports additional parameters in the checkout functions. These must also be available in the PHP SDK.

```
/**
 * Start checkout process.
 *
 * @param paymentData payment data
 * @param startWithRegistration flag indicating if the first step should be the registration if the plenigo login is used
 * @param sourceUrl URL of the page that lead to that checkout
 * @param targetUrl URL of the page the process should redirect to after successful payment
 * @param affiliateId affiliate id to associate with this checkout
 */
plenigo.checkout = function (paymentData, startWithRegistration, sourceUrl, targetUrl, affiliateId);
```

```
/**
 * Start checkout process with remote login.
 *
 * @param paymentData payment data
 * @param loginToken  login token provided
 * @param sourceUrl URL of the page that lead to that checkout
 * @param targetUrl URL of the page the process should redirect to after successful payment
 * @param affiliateId affiliate id to associate with this checkout
 */
plenigo.checkoutWithRemoteLogin = function (paymentData, loginToken, sourceUrl, targetUrl, affiliateId);
```

### PR DESCRIPTION
**Pull request by SebaSOFT**
This fixes #26

### CODE REVIEW

File: src/plenigo/builders/CheckoutSnippetBuilder.php [hunk 1]
```diff
@@ -49,14 +49,17 @@ public function __construct(Product $productToChkOut) {
      * options have been selected, this will produce a Javascript snippet of
      * code that can be used as an event on a webpage.
      *
-     * @param array $settings A map of settings to pass to the Checkout service interface.
-     * @param string $loginToken A string contianing the loginToken if obtained from the external user management
-     * @param bool $showRegisterFirst Shows a register screen if the user is not loged in
-     *
+     * @param array $settings A map of settings to pass to the Checkout service interface
+     * @param string $loginToken A string contianing the loginToken (if obtained from the external user management) or null
```
Comment (robertholfeld) [inline src/plenigo/builders/CheckoutSnippetBuilder.php]: contianing

File: src/plenigo/builders/CheckoutSnippetBuilder.php [hunk 2]
```diff
@@ -67,16 +70,47 @@ public function build($settings = array(), $loginToken = null, $showRegisterFirs
...
+            $strSurceURL = ", '" . $sourceUrl . "'";
+            if (!$showRegisterFirst && is_null($loginToken)) {
+                $strFirstParam = ", false";
+            }
+        }
+        if (!is_null($targetUrl)) {
+            PlenigoManager::get()->notice($clazz, "Target URL:" . $targetUrl);
+            $strTargetURL = ", '" . $targetUrl . "'";
+            if (is_null($sourceUrl)) {
+                $strSurceURL = ", null";
```
Comment (robertholfeld) [inline src/plenigo/builders/CheckoutSnippetBuilder.php]: Why you dont initialize $strSurceURL with ", null"; so you dont need this if condition,
and maybe $strSourceURL.

File: src/plenigo/builders/CheckoutSnippetBuilder.php (hunk 1 above)
Comment (SebaSOFT) [reply src/plenigo/builders/CheckoutSnippetBuilder.php]: Fixed

**Review by SebaSOFT — COMMENTED** [review]
..

File: src/plenigo/builders/CheckoutSnippetBuilder.php (hunk 2 above)
Comment (SebaSOFT) [reply src/plenigo/builders/CheckoutSnippetBuilder.php]: $strSourceURL is initialized to an empty string in line 84 to not send the parameter to the JavaScript function if none of the latter parameters are sent (null values).

Comment (robertholfeld) [reply src/plenigo/builders/CheckoutSnippetBuilder.php]: Its still $strSurceURL i think it should be $strSourceURL.

Comment (SebaSOFT) [reply src/plenigo/builders/CheckoutSnippetBuilder.php]: fixed

**Pull request merged**